### PR TITLE
HPCC-29035 Ecl Watch v9 errors warnings bell icon tweak

### PR DESCRIPTION
**Pull request by jeclrsg**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/Title.tsx [hunk 1]
```diff
@@ -234,8 +233,9 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
                         </Stack.Item>
                     }
                     <Stack.Item align="center">
-                        <DefaultButton href="#/log" title={nlsHPCC.ErrorWarnings} iconProps={{ iconName: log.length > 0 ? "RingerSolid" : "Ringer" }} className={btnStyles.errorsWarnings}>
-                            <span className={btnStyles.errorsWarningsCount}>{`(${log.length})`}</span>
+                        <DefaultButton href="#/log" title={nlsHPCC.ErrorWarnings} className={btnStyles.errorsWarnings}>
```
Comment (GordonSmith) [inline esp/src/src-react/components/Title.tsx:236]: We should keep the Bell Icon and either overlay the count on it or beside it where the parens used to be?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/Title.tsx [hunk 2]
```diff
@@ -87,6 +88,34 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
...
+    const [logCount, setLogCount] = React.useState(0);
+    React.useEffect(() => {
+        const errorCodes = [Level.alert, Level.critical, Level.emergency, Level.error];
+        const warningCodes = [Level.warning];
+        const errorAndWarningCodes = errorCodes.concat(warningCodes);
+
+        const logCounts = {
+            errors: log.filter(msg => errorCodes.includes(msg.level)).length,
+            warnings: log.filter(msg => warningCodes.includes(msg.level)).length,
+            other: log.filter(msg => !errorAndWarningCodes.includes(msg.level)).length,
```
Comment (GordonSmith) [inline esp/src/src-react/components/Title.tsx:100]: Minor, but `other = log.length - errors - warnings`?

File: esp/src/src-react/components/Title.tsx [hunk 3]
```diff
@@ -87,6 +88,34 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
...
+
+        if (logCounts.warnings > count) {
+            count = logCounts.warnings;
+            color = "important";
+        }
+        if (logCounts.errors > count) {
+            count = logCounts.errors;
+            color = "danger";
+        }
+
```
Comment (GordonSmith) [inline esp/src/src-react/components/Title.tsx:114]: How about:
```ts
        if (logCounts.errors > count) {
            count = logCounts.errors;
            color = "danger";
        } else if (logCounts.warnings > count) {
            count = logCounts.warnings;
            color = "important";
        } else {
            count = logCounts.other;
            color: CounterBadgeProps["color"] = "informative";
        }
```

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/Title.tsx (hunk 2 above)
Comment (jeclrsg) [reply esp/src/src-react/components/Title.tsx:100]: That's right. The CounterBadge only has 3 colors, so errors = danger, warnings = important and anything else = informative

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**